tooling/template-base/components/CourseLayout.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import { usePageContext } from '../renderer/usePageContext';
+import CourseSidebar from './CourseSidebar';
+
+export default function CourseLayout({
+  title,
+  coursePages
+}: {
+  title?: string;
+  coursePages: Array<JSX.Element>;
+}) {
+  const pageContext = usePageContext();
+  let pageQuery = 1;
+
+  const { search } = pageContext.urlParsed;
+  if (search?.page && parseInt(search.page)) {
+    pageQuery = parseInt(search.page);
+  }
+
+  const [currentPageIndex, setCurrentPage] = useState<number>(pageQuery - 1);
+  const [nextIsDisabled, setNextIsDisabled] = useState<boolean>(
+    currentPageIndex >= coursePages.length - 1
+  );
+  const [prevIsDisabled, setPrevIsDisabled] = useState<boolean>(currentPageIndex === 0);
+  const currentView = coursePages[currentPageIndex];
+
+  const handleForward = () => {
+    if (currentPageIndex < coursePages.length - 1) {
+      setCurrentPage(currentPageIndex + 1);
+    }
+  };
+
+  const handleBackward = () => {
+    if (currentPageIndex > 0) {
+      setCurrentPage(currentPageIndex - 1);
+    }
+  };
+
+  useEffect(() => {
+    setNextIsDisabled(currentPageIndex >= coursePages.length - 1);
+    setPrevIsDisabled(currentPageIndex === 0);
+  }, [currentPageIndex, coursePages.length]);
```

Comment:
I don't think any of this needs to be state or `useEffect` at all --

`const nextIsDisabled = currentPageIndex >= coursePages.length - 1;`

https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes